File: 프로그래머스/lv2/42587. 프로세스/프로세스.swift
import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    var queue = priorities
    var loc = location
    var cnt = 0
    while !queue.isEmpty {
        let p = queue.removeFirst()
        loc -= 1
        if p < queue.max() ?? 0 {
            queue.append(p)
            if loc < 0 {
                loc = queue.count-1
            }
        }else {
            cnt += 1
            if loc < 0 {
                return cnt
            }
        }
       
    }
    return cnt
}

File: 프로그래머스/lv2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    loc = location
    while priorities:
        elem = priorities.pop(0)
        loc -= 1
        if not priorities: return answer + 1
        if elem < max(priorities):
            priorities.append(elem)
            if loc < 0:
                loc = len(priorities) - 1
        else:
            answer += 1
            if loc < 0: return answer
            
    return answer